// fileManager/fileManager.js
'use strict';
var qtools = require('qtools'),
	qtools = new qtools(module),
	events = require('events'),
	util = require('util'),
	fs=require('fs');

//START OF moduleFunction() ============================================================

var moduleFunction = function(args) {
	events.EventEmitter.call(this);
	this.forceEvent = forceEvent;
	this.args = args;
	this.ping=qtools.ping;

	qtools.validateProperties({
		subject: args,
		targetScope: this, //will add listed items to targetScope
		propList: [
			{
				name: 'fileName',
				optional: false
			},
			{
				name: 'storageDirectoryPath',
				optional: false
			},
			{
				name: 'append',
				optional: true
			}
		]
	});


	var self = this,
		forceEvent = function(eventName, outData) {
			this.emit(eventName, {
				eventName: eventName,
				data: outData
			});
		};
		
	//LOCAL FUNCTIONS ====================================

	//METHODS AND PROPERTIES ====================================

	this.getItForMe=function(fileReadCallback){
		var filePath=self.storageDirectoryPath+'/'+self.fileName;

		var fileData = fs.readFile(filePath, fileReadCallback);
	}
	
	

	this.takeItAway=function(data, fileWriteCallback){
		var filePath=self.storageDirectoryPath+self.fileName;
			qtools.writeSureFile(filePath, data, {callback:fileWriteCallback, append:self.append});

	}

	//INITIALIZATION ====================================
	
	this.append=this.append?this.append:false;
	this.storageDirectoryPath=this.storageDirectoryPath.replace(/\/+$/, '')+"/";

	return this;
};

//END OF moduleFunction() ============================================================

util.inherits(moduleFunction, events.EventEmitter);
module.exports = moduleFunction;
